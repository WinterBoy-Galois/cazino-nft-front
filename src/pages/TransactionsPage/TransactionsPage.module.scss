@import '../../styles/shared';

.details-container {
  margin: 0 -0.5rem 0 -0.5rem;
  border-radius: 0;
  background-color: $blue;

  @include media-breakpoint-up(md) {
    margin: 0;
    border-radius: 8px;
  }
}

.nav {
  padding: 0 1rem;

  @include media-breakpoint-up(md) {
    padding: 0 2rem;
  }

  &__link {
    font-size: 17px;
    color: $table-ultra-light-blue;
    padding-bottom: 0.1rem;
    display: inline-block;
    margin-right: 1rem;

    &:last-of-type {
      margin: 0;
    }

    &:hover {
      color: $live-updates-tab-select-active;
      text-decoration: none;
    }

    @include media-breakpoint-up(md) {
      font-size: 20px;
      padding-bottom: 0.2rem;
      margin-right: 2rem;
    }

    &--active {
      border-bottom: 2px solid $live-updates-tab-select-active;
      color: $live-updates-tab-select-active;
    }
  }
}
