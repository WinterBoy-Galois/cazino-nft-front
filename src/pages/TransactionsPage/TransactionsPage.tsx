import { Redirect, RouteComponentProps, Router } from '@reach/router';
import React from 'react';
import { useTranslation } from 'react-i18next';
import DetailsContainer from '../../components/DetailsContainer';
import NavLink from '../../components/NavLink';
import PageContentContainer from '../../components/PageContentContainer';
import PageHeadline from '../../components/PageHeadline';
import { BetsWithData } from './components/Bets';
import { DepositsWithData } from './components/Deposits';
import { WithdrawalsWithData } from './components/Withdrawals';
import { BonusesWithData } from './components/Bonuses';
import styles from './TransactionsPage.module.scss';

const TransactionsPage: React.FC<RouteComponentProps> = () => {
  const { t } = useTranslation(['transactions']);

  return (
    <div className="container">
      <PageHeadline>{t('pageHeadline')}</PageHeadline>

      <PageContentContainer>
        <nav className={styles.nav}>
          <NavLink
            to={'withdrawals'}
            className={styles['nav__link']}
            activeClassName={styles['nav__link--active']}
          >
            WITHDRAWALS
          </NavLink>
          <NavLink
            to={'deposits'}
            className={styles['nav__link']}
            activeClassName={styles['nav__link--active']}
          >
            Deposits
          </NavLink>
          <NavLink
            to={'bets'}
            className={styles['nav__link']}
            activeClassName={styles['nav__link--active']}
          >
            Bets
          </NavLink>
          <NavLink
            to={'bonuses'}
            className={styles['nav__link']}
            activeClassName={styles['nav__link--active']}
          >
            Bonuses
          </NavLink>
        </nav>

        <DetailsContainer className={styles['details-container']}>
          <Router>
            <Redirect from={'/'} to={'withdrawals'} noThrow />
            <WithdrawalsWithData path={'withdrawals'} />
            <DepositsWithData path={'deposits'} />
            <BetsWithData path={'bets'} />
            <BonusesWithData path={'bonuses'} />
          </Router>
        </DetailsContainer>
      </PageContentContainer>
    </div>
  );
};

export default TransactionsPage;
