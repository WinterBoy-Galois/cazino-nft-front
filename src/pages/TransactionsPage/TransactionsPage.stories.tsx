import React from 'react';

import TransactionsPage from '.';

export default {
  title: 'Components/TransactionsPage',
  component: TransactionsPage,
};

export const Default = () => <TransactionsPage />;
