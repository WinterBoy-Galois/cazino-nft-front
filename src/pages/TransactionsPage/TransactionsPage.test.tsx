import React from 'react';
import { render } from '@testing-library/react';

import TransactionsPage from '.';

describe('TransactionsPage', () => {
  it('should match snapshot', () => {
    // Arrange

    // Act
    const container = render(<TransactionsPage />);

    // Assert
    expect(container).toMatchSnapshot();
  });
});
