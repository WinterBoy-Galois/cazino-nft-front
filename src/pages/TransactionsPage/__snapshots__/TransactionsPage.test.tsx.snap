// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionsPage should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <h1
          class="headline"
        >
          transactions
        </h1>
        <div
          class="container "
        >
          <nav
            class="nav"
          >
            <a
              class="nav__link"
              href="/withdrawals"
            >
              WITHDRAWALS
            </a>
            <a
              class="nav__link"
              href="/deposits"
            >
              Deposits
            </a>
            <a
              class="nav__link"
              href="/bets"
            >
              Bets
            </a>
            <a
              class="nav__link"
              href="/bonuses"
            >
              Bonuses
            </a>
          </nav>
          <div
            class="details-container container container--light"
          >
            <div
              style="outline: none;"
              tabindex="-1"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <h1
        class="headline"
      >
        transactions
      </h1>
      <div
        class="container "
      >
        <nav
          class="nav"
        >
          <a
            class="nav__link"
            href="/withdrawals"
          >
            WITHDRAWALS
          </a>
          <a
            class="nav__link"
            href="/deposits"
          >
            Deposits
          </a>
          <a
            class="nav__link"
            href="/bets"
          >
            Bets
          </a>
          <a
            class="nav__link"
            href="/bonuses"
          >
            Bonuses
          </a>
        </nav>
        <div
          class="details-container container container--light"
        >
          <div
            style="outline: none;"
            tabindex="-1"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
